netty/src/main/java/io/grpc/netty/NettyServerStream.java

```diff
@@ -53,6 +58,13 @@
   private final NettyServerHandler handler;
   private final Http2Stream http2Stream;
   private final WriteQueue writeQueue;
+  private final Attributes attributes;
+
+  private static Attributes.Key<SocketAddress> REMOTE_ADDR_KEY =
+      Attributes.Key.named("remote-addr");
```

Comment:
Currently `Key.name` is used as the key in the attributes map. I think we should use the `Key` object instead, so that `Key.name` is only used in case `toString()` is needed and `Key`s won't accidentally collide by having the same name. That would require the `Key` object to be public, which I think makes sense.